megamek/src/megamek/common/EntityListFile.java

```diff
@@ -268,6 +274,13 @@ public static String getLocString(Entity entity, int indentLvl) {
                     if (CriticalSlot.TYPE_EQUIPMENT == slot.getType()) {
                         mount = slot.getMount();
                     }
+                    
+                    // if the "equipment" is a weapons bay, 
+                    // then let's make a note of it
+                    if(entity.usesWeaponBays() && 
+                            slot.getMount().getBayAmmo() != null) {
```

Comment:
`Mounted` initializes `bayAmmo` to an empty vector. `getBayAmmo()` is never going to return `null`, so this map is going to contain every `Mounted` slot up to the current one. I can't think of a way to test for a weapon bay other than `slot.getMount() instanceof BayWeapon`.

This also relies on the bay critical slot always coming before the ammo. This may be a valid assumption, but I'm not certain of that.

Reply:
Probably checking getBayAmmo().length ought to do it then. I'll look at the possibility of a bay coming after the ammo tonight. Seems unlikely from looking at some random blk files, but worth double checking.

Reply:
I added a weapon before a bay in a dropship .blk file and it actually auto-generates a bay in the previous slot if there isn't one already.